adminapi/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -613,6 +619,257 @@ public Message getServerInfo() throws IOException, NoSuchAlgorithmException, Inv
+   * @param description Description for this access key.
+   * @param comment Deprecated: use description instead.
+   * @param expiration Expiry time , Example : 2023-12-02T15:04:05Z.
+   * @return Service account info for the specified accessKey.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws IOException thrown to indicate I/O error on MinIO REST operation.
+   * @throws InvalidCipherTextException thrown to indicate data cannot be encrypted/decrypted.
+   */
+  public Credentials addServiceAccount(
+      @Nonnull String targetUser,
+      @Nullable String name,
+      @Nonnull String secretKey,
+      @Nonnull String accessKey,
+      @Nullable String policy,
+      @Nullable String description,
+      @Nullable ZonedDateTime expiration,
+      @Nullable String comment)
+      throws NoSuchAlgorithmException, InvalidKeyException, IOException,
+          InvalidCipherTextException {
+    if (targetUser == null || targetUser.isEmpty()) {
+      throw new IllegalArgumentException("target user must be provided");
+    }
+    if (targetUser.length() > 32) {
+      throw new IllegalArgumentException("name must not be longer than 32 characters");
+    }
+    if (description != null && description.length() > 256) {
+      throw new IllegalArgumentException("description must be at most 256 bytes long");
+    }
+    if (accessKey == null || accessKey.isEmpty()) {
+      throw new IllegalArgumentException("access key must be provided");
+    }
+    if (secretKey == null || secretKey.isEmpty()) {
+      throw new IllegalArgumentException("secret key must be provided");
+    }
+    Map<String, Object> serviceAccount = new HashMap<>(8);
+    serviceAccount.put("targetUser", targetUser);
+    serviceAccount.put("accessKey", accessKey);
+    if (name != null && !name.isEmpty()) {
+      serviceAccount.put("name", name);
+    }
+    serviceAccount.put("secretKey", secretKey);
+    if (policy != null && !policy.isEmpty()) {
+      Map<String, Object> policyMap = OBJECT_MAPPER.readValue(policy, Map.class);
+      serviceAccount.put("policy", policyMap);
+    }
+    if (description != null && !description.isEmpty()) {
+      serviceAccount.put("description", description);
+    }
+    if (comment != null && !comment.isEmpty()) {
+      serviceAccount.put("comment", comment);
+    }
+    String expirationFormat = null;
+    if (expiration != null) {
+      expirationFormat = expiration.format(Time.EXPIRATION_DATE_FORMAT);
+      serviceAccount.put("expiration", expirationFormat);
+    }
+    Credentials creds = getCredentials();
+    try (Response response =
+        execute(
+            Method.PUT,
+            Command.ADD_SERVICE_ACCOUNT,
+            ImmutableMultimap.of("accessKey", accessKey),
+            Crypto.encrypt(creds.secretKey(), OBJECT_MAPPER.writeValueAsBytes(serviceAccount)))) {
```

Comment:
You would need to add `AddServiceAccountResp` class and do XML unmarshaling the response and return `credentials` from the parsed result.